command/chain_watch.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/ethereum/go-ethereum/command/flagset"
+	"github.com/ethereum/go-ethereum/command/server/proto"
+	"github.com/ethereum/go-ethereum/core"
+)
+
+// ChainWatchCommand is the command to group the peers commands
+type ChainWatchCommand struct {
+	*Meta2
+}
+
+// Help implements the cli.Command interface
+func (c *ChainWatchCommand) Help() string {
+	return ``
+}
+
+func (c *ChainWatchCommand) Flags() *flagset.Flagset {
+	flags := c.NewFlagSet("chain watch")
+
+	return flags
+}
+
+// Synopsis implements the cli.Command interface
+func (c *ChainWatchCommand) Synopsis() string {
+	return ""
+}
+
+// Run implements the cli.Command interface
+func (c *ChainWatchCommand) Run(args []string) int {
+	flags := c.Flags()
+	if err := flags.Parse(args); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	borClt, err := c.BorConn()
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	sub, err := borClt.ChainWatch(context.Background(), &proto.ChainWatchRequest{})
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		<-signalCh
+		sub.CloseSend()
+	}()
+
+	for {
+		msg, err := sub.Recv()
+		if err != nil {
+			// if err == EOF if finished on the other side
+			c.UI.Output(err.Error())
+			break
+		}
+		if msg.Type == core.Chain2HeadCanonicalEvent {
+			out := fmt.Sprintf("Block Added : %v", msg.Newchain)
```

Comment:
This does not look nice. It would be better to create a print function that shows the event:
```
func printEvent(event *proto....) string {
   var out string
   if msg.Type == core.Chain2HeadCanonicalEvent {
       out = fmt.Sprintf("Block Added : %v", msg.Newchain)
   } else if ... {

   }
   return out
}
```
and then:
```
msg, err := sub.Recv()
c.UI.Output(printEvent(msg))
```